core/src/main/java/org/bitcoinj/core/Wallet.java

```diff
@@ -1617,7 +1615,7 @@ public void receivePending(Transaction tx, @Nullable List<Transaction> dependenc
             // If this tx spends any of our unspent outputs, mark them as spent now, then add to the pending pool. This
             // ensures that if some other client that has our keys broadcasts a spend we stay in sync. Also updates the
             // timestamp on the transaction and registers/runs event listeners.
-            commitTx(tx);
+            commitTx(tx, false);
```

Comment:
There's still no way to miss it:

* When you come up again, if it's unconfirmed, it will normally be sent to you from the mempool of the peers you connect to
* If it gets confirmed in the mean time, you will scan it from the blockchain in the normal way

The only way to miss it completely, is if it didn't propagate in the network and never confirms, which is a bad situation anyway.